src/main/java/org/generationcp/middleware/dao/dms/StockDao.java

```diff
@@ -292,5 +305,119 @@ public void replaceExperimentStocks(final Integer oldStockId, final Integer newS
+				+ "  IF(COUNT(DISTINCT ifnull(l.scaleid, 'null')) = 1, IFNULL(c.name,'-'), 'Mixed') AS unit ");
+
+			final String entryClause = ",MAX(IF(cvterm_variable.name = '%s', sp.value, NULL)) AS '%s',"
+				+ " MAX(IF(cvterm_variable.name = '%s', sp.stockprop_id, NULL)) AS '%s',"
+				+ " MAX(IF(cvterm_variable.name = '%s', sp.type_id, NULL)) AS '%s' ,"
+				+ " MAX(IF(cvterm_variable.name = '%s', sp.value, NULL)) AS '%s' ";
+
+			for (final MeasurementVariable entryDescriptor : studyEntrySearchDto.getEntryDescriptors()) {
+				final String entryName = entryDescriptor.getName();
+				sqlQuery.append(String.format(entryClause, entryName, entryName, entryName, entryName + "_PropertyId",
+					entryName, entryName + "_variableId",
+					entryName, entryName + "_value"
+				));
+			}
+
+			sqlQuery.append(" FROM stock s "
+				+ "       LEFT JOIN ims_lot l ON l.eid = s.dbxref_id and l.status = " + LotStatus.ACTIVE.getIntValue()
+				+ "       LEFT JOIN ims_transaction it ON l.lotid = it.lotid "
+				+ "       LEFT JOIN cvterm c ON c.cvterm_id = l.scaleid "
+				+ "       LEFT JOIN stockprop sp ON sp.stock_id = s.stock_id "
+				+ "       LEFT JOIN cvterm cvterm_variable ON cvterm_variable.cvterm_id = sp.type_id "
+				+ "WHERE s.project_id = :studyId ");
+
+
+			if (studyEntrySearchDto.getFilter()!=null) {
+				if (studyEntrySearchDto.getFilter().getEntryNumbers()!=null && !studyEntrySearchDto.getFilter().getEntryNumbers().isEmpty()) {
+					sqlQuery.append(" AND s.uniquename in (:entryNumbers)" );
+				}
+				if (studyEntrySearchDto.getFilter().getEntryIds()!=null && !studyEntrySearchDto.getFilter().getEntryIds().isEmpty()) {
+					sqlQuery.append(" AND s.stock_id in (:entryIds)" );
+				}
+			}
+			sqlQuery.append(" GROUP BY s.stock_id ");
+			GenericDAO.addPageRequestOrderBy(sqlQuery, pageable);
+
+			final SQLQuery query = this.getSession().createSQLQuery(sqlQuery.toString());
+			query.addScalar("entryId", new IntegerType());
+			query.addScalar("entryNumber", new IntegerType());
+			query.addScalar("gid", new IntegerType());
+			query.addScalar("designation", new StringType());
+			query.addScalar("entryCode", new StringType());
+			query.addScalar("lots", new IntegerType());
+			query.addScalar("available", new StringType());
+			query.addScalar("unit", new StringType());
+			for (final MeasurementVariable entryDescriptor : studyEntrySearchDto.getEntryDescriptors()) {
+				final String entryName = entryDescriptor.getName();
+				query.addScalar(entryName + "_propertyId", new IntegerType());
+				query.addScalar(entryName + "_variableId", new IntegerType());
+				query.addScalar(entryName + "_value", new StringType());
+			}
+
+			query.setParameter("studyId", studyEntrySearchDto.getStudyId());
+			if (studyEntrySearchDto.getFilter()!=null) {
+				if (studyEntrySearchDto.getFilter().getEntryNumbers()!=null && !studyEntrySearchDto.getFilter().getEntryNumbers().isEmpty()) {
```

Comment:
Can be simplified as: 	
`if (!CollectionUtils.isEmpty(studyEntrySearchDto.getFilter().getEntryNumbers())) {`